### DISCUSSION TITLE
App startup

### DISCUSSION THREAD

**Discussion by Darrshan-Sankar** [rajkumardusad/jitcompiler#9] [General]
I have set this up in a windows system, built the docker container too and the container is on. When i run npm start, I get stuck up with this:
![image](https://github.com/rajkumardusad/jitcompiler/assets/82897100/adc063bd-7ac2-49c6-9edc-b0fb1796145d)
After this even if it runs longer, I get nothing.

Any support on how to proceed further?

**Comment by rajkumardusad** [comment]
> I have set this up in a windows system, built the docker container too and the container is on. When i run npm start, I get stuck up with this: ![image](https://private-user-images.githubusercontent.com/82897100/346316771-adc063bd-7ac2-49c6-9edc-b0fb1796145d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzNDI5NDcsIm5iZiI6MTcyMDM0MjY0NywicGF0aCI6Ii84Mjg5NzEwMC8zNDYzMTY3NzEtYWRjMDYzYmQtN2FjMi00OWM2LTllZGMtYjBmYjE3OTYxNDVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDA4NTcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxNzhhYzA4Njc4NTVmYzAyZTZmZmI0OThjNmViOTI0Nzg5Nzk5MWQ5YjZiM2M3ZGQ1Y2I5YzU0NDY3N2NjNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fqXzwSdi9QqQ1--kTwI6H7fBA4SZEsVIrvbBqc0SYrM) After this even if it runs longer, I get nothing.
> 
> Any support on how to proceed further?

Is there are any error message? If there is no any error then run http://localhost:3000 in browser it will works

**Reply by Darrshan-Sankar** [reply]
Yes there is, but this is an awesome project so i wanted to put this in discussion.

Here is the message:
![image](https://github.com/rajkumardusad/jitcompiler/assets/82897100/32a2b9c6-b580-4c75-b73d-d0bd68dc5b29)

The above message has come in the console, when i tried to execute the code

**Reply by Darrshan-Sankar** [reply]
This is for additional clarity
![image](https://github.com/rajkumardusad/jitcompiler/assets/82897100/d5e2ea66-3ac7-4d7b-9672-21bba05aeed4)

**Reply by rajkumardusad** [reply]
This error is only for pwa, the compiler should be working without any issue. I will check this error any update you soon. Thanks for informing us.

**Reply by rajkumardusad** [reply]
For the time being you disable the service_worker.js if it is not working. It will not affect the project.

**Reply by Darrshan-Sankar** [reply]
I was reinspecting the docker. When i tried to rebuild the docker, I found this error:

1.500 npm ERR! code EBADENGINE
1.501 npm ERR! engine Unsupported engine
1.501 npm ERR! engine Not compatible with your version of node/npm: npm@10.8.1
1.501 npm ERR! notsup Not compatible with your version of node/npm: npm@10.8.1
1.501 npm ERR! notsup Required: {"node":"^18.17.0 || >=20.5.0"}
1.501 npm ERR! notsup Actual:   {"npm":"8.19.4","node":"v16.20.2"}
1.502
1.502 npm ERR! A complete log of this run can be found in:
1.502 npm ERR!     /root/.npm/_logs/2024-07-09T16_25_20_772Z-debug-0.log
------
Dockerfile:31
--------------------
  29 |     RUN apt-get install nodejs -y
  30 |     # Install bhailang
  31 | >>> RUN npm install -g npm
  32 |     RUN npm i -g bhailang
  33 |     # Install python
--------------------
ERROR: failed to solve: process "/bin/bash -c npm install -g npm" did not complete successfully: exit code: 1

And this fixed for me, by updating this in docker:
 Install nodejs (Node.js 16.x)
RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
RUN apt-get install nodejs -y

to:
Install nodejs (Node.js 18.x)
RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
RUN apt-get install nodejs -y

**Reply by Darrshan-Sankar** [reply]
Even after this, every set-up performed as given, but unable to execute the code. Support requested on this.

**Comment by Darrshan-Sankar** [comment]
Could you please be more specific, because, even after i commented out all the lines in service_worker.js, the code didn't get executed in the  and display the result in the output section.

should i provide my setup.bat which i used to perform the same?
